### DISCUSSION TITLE
Preserving grouped tags from Darktable in Immich

### DISCUSSION THREAD

**Discussion by ontam** [immich-app/immich#14391] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi, is there any way to import photos with external tags created in Darktable into Immich while preserving the tag grouping structure?
Darktable uses the level_1|level_2 format for grouping tags, whereas Immich seems to use the level_1/level_2 format.

https://docs.darktable.org/usermanual/4.0/en/module-reference/utility-modules/shared/tagging/

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Have you tried it? Immich already handles darktable's format.

**Reply by ontam** [reply]
Yes, I tried it, and the tags were successfully imported, but the groups were split into multiple individual tags. For example, from the tag level_1|level_2, I ended up with two separate tags: level_1 and level_2. I will try again.

**Reply by ontam** [reply]
Anything new about my problem?

**Comment by ontam** [comment]
I updated immich and tried again, still same problem:

Tag in darktable, 3 levels:
![darktable](https://github.com/user-attachments/assets/4c0a8697-2680-4697-b9ab-b5a00e62b58e)


Same photo in immich with separated group tag (3 levels) into 3 single tags: 
<img width="425" alt="immich" src="https://github.com/user-attachments/assets/ba115d8d-0eff-4caa-885d-2abe8296ec10">

Edit: 
screen from immich tags explorer
<img width="148" alt="immich_explorer" src="https://github.com/user-attachments/assets/092a760e-0096-4640-b9a9-ac7419e82ffd">

**Comment by steph72** [comment]
So my bug report was closed as a duplicate, but I still think it has some merit since it analyzes the source of the problem, which should make fixing the bug a little easier:

(and not wanting to step on anyone's toes, but I really think this is a bug, not a feature request)

Darktable stores its tags inside two nodes:  A "<lr:hierarchicalSubject>" node like so:
```
   <lr:hierarchicalSubject>
    <rdf:Bag>
     <rdf:li>darktable|format|orf</rdf:li>
     <rdf:li>unterwegs|föhr|2024</rdf:li>
    </rdf:Bag>
   </lr:hierarchicalSubject>
```

and a "<dc:subject>"  node like so:
```
   <dc:subject>
    <rdf:Bag>
     <rdf:li>2024</rdf:li>
     <rdf:li>darktable</rdf:li>
     <rdf:li>format</rdf:li>
     <rdf:li>föhr</rdf:li>
     <rdf:li>orf</rdf:li>
     <rdf:li>unterwegs</rdf:li>
    </rdf:Bag>
   </dc:subject>
```

The hierarchy is indicated in the "<lr:hierarchicalSubject>" node with the "|" sign as path delimiter.

Immich seems to only import the "<dc:subject>" node.

**Comment by pedvide** [comment]
I have exactly the same problem importing my darktable photos into immich. It would be great to actually support the tag hierarchy!

**Reply by pedvide** [reply]
I actually found the problem, at least in my case: I was adding jpegs to my external library. Darktable by default exports the list of tags, without hierarchy. Going to the preferences of the [export module](https://docs.darktable.org/usermanual/4.6/en/module-reference/utility-modules/shared/export/), I added hierarchical tags. Now after reexporting, the hierarchical tags are in the jpeg, and immich reads them correctly
